src/xunit.runner.uap/StandardUapVisitor.cs

```diff
@@ -0,0 +1,190 @@
+{
+    public class StandardUapVisitor : XmlTestExecutionVisitor
+    {
+        string assemblyName;
+        readonly ConcurrentDictionary<string, ExecutionSummary> completionMessages;
+        readonly StreamWriter log;
+        readonly bool showProgress;
+        readonly bool failSkips;
+
+        public StandardUapVisitor(XElement assemblyElement,
+                                     Func<bool> cancelThunk,
+                                     StreamWriter log,
+                                     ConcurrentDictionary<string, ExecutionSummary> completionMessages,
+                                     bool showProgress,
+                                     bool failSkips)
+            : base(assemblyElement, cancelThunk)
+        {
+            this.completionMessages = completionMessages;
+            this.log = log;
+            this.showProgress = showProgress;
+            this.failSkips = failSkips;
+        }
+
+        public ExecutionSummary ExecutionSummary 
+        {
+            get 
+            {
+                ExecutionSummary summary;
+                if (completionMessages.TryGetValue(assemblyName, out summary))
+                {
+                    return summary;
+                }
+
+                return new ExecutionSummary();
+            }
+        }
+
+        protected override bool Visit(ITestAssemblyStarting assemblyStarting)
+        {
+            assemblyName = Path.GetFileNameWithoutExtension(assemblyStarting.TestAssembly.Assembly.AssemblyPath);
+
+            log.WriteLine($"Starting:    {assemblyName}");
+
+            return base.Visit(assemblyStarting);
+        }
+
+        protected override bool Visit(ITestAssemblyFinished assemblyFinished)
+        {
+            // Base class does computation of results, so call it first.
+            var result = base.Visit(assemblyFinished);
+
+            log.WriteLine($"Finished:    {assemblyName}");
+
+            completionMessages.TryAdd(assemblyName, new ExecutionSummary
+            {
+                Total = assemblyFinished.TestsRun,
+                Failed = !failSkips ? assemblyFinished.TestsFailed : assemblyFinished.TestsFailed + assemblyFinished.TestsSkipped,
+                Skipped = !failSkips ? assemblyFinished.TestsSkipped : 0,
+                Time = assemblyFinished.ExecutionTime,
+                Errors = Errors
+            });
+
+            return result;
+        }
+
+        protected override bool Visit(ITestFailed testFailed)
+        {
+            log.WriteLine($"   {XmlEscape(testFailed.Test.DisplayName)} [FAIL]");
+            log.WriteLine($"      {ExceptionUtility.CombineMessages(testFailed).Replace(Environment.NewLine, Environment.NewLine + "      ")}");
+
+            WriteStackTrace(ExceptionUtility.CombineStackTraces(testFailed));
+
+            return base.Visit(testFailed);
+        }
+
+        protected override bool Visit(ITestPassed testPassed)
+        {
+            return base.Visit(testPassed);
```

Comment:
>return base.Visit(testPassed); [](start = 12, length = 30)

do we need to count the failed and passed test to produce the summary at the end of the run?

Reply:
Nope :), xunit will handle that for us, the base class will take care of it and then when test assembly is finished executing we will get the results on the Visit method.

https://github.com/dotnet/buildtools/pull/1667/files#diff-ee71c3a2de1958c609bbdc6fac71d0adR58